config/v3_4_experimental/types/config_test.go

```diff
@@ -0,0 +1,230 @@
+		in  Config
+		out error
+		at  path.ContextPath
+	}{
+		// test 0: file conflicts with systemd dropin file, error
+		{
+			in: Config{
+				Ignition{},
+				KernelArguments{},
+				Passwd{},
+				Storage{
+					Files: []File{
+						{
+							Node: Node{Path: "/etc/systemd/system/foo.service.d/bar.conf"},
+						},
+					},
+				},
+				Systemd{
+					Units: []Unit{
+						{Name: "foo.service", Dropins: []Dropin{{Name: "bar.conf"}}},
+					},
+				},
+			},
+			out: errors.ErrFileConflictsSystemd,
+			at:  path.New("json", "storage", "files", 0, "path"),
+		},
+		// test 1: file conflicts with systemd unit, error
+		{
+			in: Config{
+				Ignition{},
+				KernelArguments{},
+				Passwd{},
+				Storage{
+					Files: []File{
+						{
+							Node: Node{Path: "/etc/systemd/system/foo.service"},
+						},
+					},
+				},
+				Systemd{
+					Units: []Unit{
+						{
+							Name:     "foo.service",
+							Contents: util.StrToPtr("[Foo]\nQux=Bar"),
+							Enabled:  util.BoolToPtr(true),
+						},
+					},
+				},
+			},
+			out: errors.ErrFileConflictsSystemd,
+			at:  path.New("json", "storage", "files", 0, "path"),
+		},
+		// test 2: directory conflicts with systemd dropin file, error
+		{
+			in: Config{
+				Ignition{},
+				KernelArguments{},
+				Passwd{},
+				Storage{
+					Directories: []Directory{
+						{
+							Node: Node{Path: "/etc/systemd/system/foo.service.d/bar.conf"},
+						},
+					},
+				},
+				Systemd{
+					[]Unit{
+						{Name: "foo.service", Dropins: []Dropin{{Name: "bar.conf"}}},
+					},
+				},
+			},
+			out: errors.ErrDirConflictsSystemd,
+			at:  path.New("json", "storage", "directories", 0, "path"),
+		},
+		// test 3: directory conflicts with systemd unit, error
+		{
+			in: Config{
+				Ignition{},
+				KernelArguments{},
+				Passwd{},
+				Storage{
+					Directories: []Directory{
+						{
+							Node: Node{Path: "/etc/systemd/system/foo.service"},
+						},
+					},
+				},
+				Systemd{
+					[]Unit{
+						{
+							Name:     "foo.service",
+							Contents: util.StrToPtr("[foo]\nQux=Baz"),
+							Enabled:  util.BoolToPtr(true),
+						},
+					},
+				},
+			},
+			out: errors.ErrDirConflictsSystemd,
+			at:  path.New("json", "storage", "directories", 0, "path"),
+		},
+		// test 4: link conflicts with systemd dropin file, error
+		{
+			in: Config{
+				Ignition{},
+				KernelArguments{},
+				Passwd{},
+				Storage{
+					Links: []Link{
+						{
+							Node:          Node{Path: "/etc/systemd/system/foo.service.d/bar.conf"},
+							LinkEmbedded1: LinkEmbedded1{Target: util.StrToPtr("/qux.conf")},
+						},
+					},
+				},
+				Systemd{
+					[]Unit{
+						{Name: "foo.service", Dropins: []Dropin{{Name: "bar.conf"}}},
+					},
+				},
+			},
+			out: errors.ErrLinkConflictsSystemd,
+			at:  path.New("json", "storage", "links", 0, "path"),
+		},
+		// test 5: file does not conflict with systemd unit, no error
+		{
+			in: Config{
+				Ignition{},
+				KernelArguments{},
+				Passwd{},
```

Comment:
Unnecessary.

Reply:
https://github.com/coreos/ignition/actions/runs/3191845930/jobs/5208635971#step:5:30

Reply:
This has been fixed in the latest push.